packages/signer/src/Signer.js

```diff
@@ -0,0 +1,273 @@
+            ],
+        });
+        signedAttrs.push(new pkijs.Attribute({
+            type: oids.signingCertificateV2,
+            values: [signingCertificateV2],
+        }));
+        return signedAttrs;
+    }
+
+    /**
+     * Obtain the unsigned attributes.
+     * @returns {pkijs.Attribute[]}
+     */
+    async obtainUnsignedAttributes(signature) {
+        return [];
+    }
+
+    /**
+     * @param {Buffer} pdfBuffer
+     * @param {Date | undefined} signingTime
+     * @returns {Promise<Buffer>}
+     */
+    async sign(pdfBuffer, signingTime = undefined) {
+        if (!(pdfBuffer instanceof Buffer)) {
+            throw new SignPdfError(
+                'PDF expected as Buffer.',
+                SignPdfError.TYPE_INPUT,
+            );
+        }
+
+        // Get signature and hash algorithms
+        this.signAlgorithm = await this.getSignAlgorithm();
+        this.hashAlgorithm = await this.getHashAlgorithm();
+        // Get a crypto extension
+        this.crypto = this.getCrypto();
+        // Get the signing (and verification) certificates
+        const certificates = await this.obtainCertificates();
+        const signCert = certificates[0];
+        // Obtain the private key used for signing
+        const key = await this.obtainKey();
+
+        // Creation of the CMS Signed Data structure (see [PDF - 12.8.3.3] and [PAdES - 4.1])
+        // Setup signer info (see [CMS - 5.3])
+        const signerInfo = new pkijs.SignerInfo({
+            version: 1,
+            sid: new pkijs.IssuerAndSerialNumber({
+                issuer: signCert.issuer,
+                serialNumber: signCert.serialNumber,
+            }),
+            signedAttrs: new pkijs.SignedAndUnsignedAttributes({
+                type: 0,
+                attributes: await this.obtainSignedAttributes(signingTime, pdfBuffer, signCert),
+            }),
+        });
+
+        // Initialize CMS Signed Data structure (see [CMS - 5.1]) and sign it
+        const cmsSignedData = new pkijs.SignedData({
+            version: 1,
+            encapContentInfo: new pkijs.EncapsulatedContentInfo({
+                eContentType: oids.data, // No actual econtent here, as we sign in detached mode
+            }),
+            signerInfos: [signerInfo],
+            certificates,
+        });
+
+        // Sign in detached mode. That's what Adobe.PPKLite expects for subfilters
+        // adbe.pkcs7.detached and ETSI.CAdES.detached.
+        await cmsSignedData.sign(key, 0, this.hashAlgorithm, undefined, this.crypto);
+
+        // Append the unsigned attributes, if there are any
+        const unsignedAttrs = await this.obtainUnsignedAttributes(signerInfo.signature.getValue());
+        if (unsignedAttrs.length > 0) {
+            signerInfo.unsignedAttrs = new pkijs.SignedAndUnsignedAttributes({
+                type: 1,
+                attributes: unsignedAttrs,
+            });
+        }
```

Comment:
```suggestion
        // if (unsignedAttrs.length > 0) {
        //     signerInfo.unsignedAttrs = new pkijs.SignedAndUnsignedAttributes({
        //         type: 1,
        //         attributes: unsignedAttrs,
        //     });
        // }
```
This can be commented out for now (I will reintroduce it in a follow-up PR) and should fix the coverage failure.